File: frontend/src/types/LearningRecord.ts
export interface LearningRecord {
    id: number;
    title: string;
    content: string;
    studyMinutes: number;
    createdAt: string;
    updatedAt: string;
}

// フォーム入力用の型（作成時に使用）
export type LearningRecordInput = Omit<LearningRecord, 'id' | 'createdAt' | 'updatedAt'>;

File: backend/src/main/java/com/example/learningTracker/model/LearningRecord.java
// backend/src/main/java/com/example/learningTracker/model/LearningRecord.java

package com.example.learningTracker.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class LearningRecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String title;
    private String content;
    private Integer studyMinutes;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}